aten/src/ATen/native/mps/OperationUtils.h

```diff
@@ -64,7 +64,7 @@ class Placeholder {
  public:
   Placeholder() : _placeholder(nullptr), _value(nullptr), _tensor(Tensor()) {}
   Placeholder(MPSGraphTensor* mpsGraphTensor) : _placeholder(mpsGraphTensor), _value(nullptr), _tensor(Tensor()) {}
-  Placeholder(MPSGraphTensor* mpsGraphTensor, const Tensor& self, MPSShape *mpsShape = nullptr, bool gatherTensorData = true);
+  Placeholder(MPSGraphTensor* mpsGraphTensor, const Tensor& self, MPSShape *mpsShape = nullptr, bool gatherTensorData = true, MPSDataType* dataType = nullptr);
```

Comment:
Why pass by pointer? You could just declare with default value like `MPSDataType dataType = MPSDataTypeInvalid` and in the code check `if (dataType != MPSDataTypeInvalid) ...`.

Reply:
Thanks - I didn't know there is a MPSDataTypeInvalid for MPSDataType

Reply:
Updated